zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt

```diff
@@ -356,22 +356,21 @@ class ZakenApiPlugin(
                 maxDurationInDays = maxDurationInDays
             )
 
-            val existingHerseltermijn = zaakHersteltermijnRepository.findByZaakUrlAndEndDateIsNull(zaakUrl)
-            check(existingHerseltermijn == null || existingHerseltermijn != hersteltermijn) { "Hersteltermijn already exists for zaak '$zaakUrl'. " }
+            require(zaakHersteltermijnRepository.findByZaakUrlAndEndDateIsNull(zaakUrl) == null) {
```

Comment:
If we want to be on the safe side to prevent data corruption, PostgreSQL supports partial (conditional) constraints on indexes:
https://www.postgresql.org/docs/current/indexes-partial.html

Reply:
Not sure what you're asking here

Reply:
The code is checking if a hersteltermijn already exists, but that could lead to the same type of concurrency/multi-instance issues we faced for documents, which might lead to multiple active hersteltermijnen.

To solve this, you could add a 'unique index' to Postgres like this:
`CREATE UNIQUE INDEX uq_hersteltermijn ON zaak_hersteltermijn (zaak_url, end_date)
    WHERE end_date IS NULL;`